### PR TITLE
Code Refactoring

### PR DESCRIPTION
**Pull request by AnastasiaBlack**
- Changed the response when we try to retrieve all the groups of the user and the user doesn't have any: now the method returns an empty collection instead of 404 "Not found" with exception.

- Changed URLs according to the product owner's vision.

### CODE REVIEW

**Comment by svoronin** [comment]
Can you update documentation according to those changes please?

**Comment by AnastasiaBlack** [comment]
Yes, sure! We've discussed it with Dima and he said that he's going to do that as the documentation is his task for this Sprint, according to jira: )

**Comment by svoronin** [comment]
All right

File: src/main/java/com/softserve/teamproject/controller/UserProfileController.java [hunk 1]
```diff
@@ -36,11 +36,11 @@ public User getUserProfile(Principal principal) {
   }
 
   /**
-   * Get groups from user location
+   * Get groups from the authenticated user location
    * @param principal authorized user
    * @return groups in user location
    */
-  @RequestMapping(value = "/user/location/groups", method = RequestMethod.GET)
+  @RequestMapping(value = "/groups/locations/my", method = RequestMethod.GET)
```
Comment (Crutovel) [inline src/main/java/com/softserve/teamproject/controller/UserProfileController.java]: I think /locations/my/groups  would be more restful endpoint

Comment (AnastasiaBlack) [inline src/main/java/com/softserve/teamproject/controller/UserProfileController.java]: I'm not sure about that, as our Product owner said that he didn't like it when we referred to the searched items like that. He wanted us to keep the following structure: "requested items"/"filter"...Any other ideas?

Comment (Crutovel) [inline src/main/java/com/softserve/teamproject/controller/UserProfileController.java]: he didn't like when we have unnecessary  /<USER>/location/groups.
Restful way is  URI way.   We dont have any identifier in /user/location/groups.
But my option has one /locations/{MY}/groups

**Pull request merged**